grovel/grovel.lisp

```diff
@@ -749,12 +862,17 @@ string."
               (named-library-name
                 (let ((*package* (find-package :keyword))
                       (*read-eval* nil))
-                  (read-from-string lib-soname))))
+                  (read-from-string lib-soname)))
+              (search-file (if cached-lib
+                               `(system-relative-pathname
+                                 ,(component-name system)
+                                 ,(pathname-directory-pathname cached-lib))
+                               (directory-namestring lib-file))))
```

Comment:
This is the part that makes sure that we can use this file portably. If we have cached the library then put the path to the library relative to the system (as it must be in the hierarchy). We use `uiop`'s `system-relative-pathname` to make sure this is portable and robust